359 - Changed date inputs to support ISO dates

clarkepeterf opened the pull request.

kamerynB commented:
@clarkepeterf Can you make 2 updates to the unit tests? 

1. [getYearsFromFacetValues](https://github.com/project-lux/lux-frontend/blob/359-support-iso-dates/client/src/test/unit/lib/facets/dateParser.spec.ts#L48) which requires the mock data to be updated so that the format of BCE dates are `-YYYYYY`. 

2. The parameter for [getYearFromSingleFacetValue](https://github.com/project-lux/lux-frontend/blob/359-support-iso-dates/client/src/test/unit/lib/facets/dateParser.spec.ts#L57) on line 57 should be `-002017-10-20T00:00:00`.

clarkepeterf commented:
@kamerynB I made those updates as well as:

1. Additional date formatting we talked about via Teams
2. Removed `advancedSearch/DateInput` which was an unused component
3. Removed `dateParser.formatYear()` and its unit test, as it was an unused function
